From a3c2ac6a18ef6a91fcbbe6027d7d07f7a15a08f6 Mon Sep 17 00:00:00 2001
From: Alan Carvalho de Assis <acassis@gmail.com>
Date: Thu, 24 Sep 2026 09:50:16 -0300
Subject: [PATCH] zbuffer.h: make the default allocator wrappers static inline

gl_malloc(), gl_zalloc() and gl_free() are defined in the public
<zbuffer.h> as plain static functions when TGL_FEATURE_CUSTOM_MALLOC is
0.  Every application that includes the header but does not call them
then gets three -Wunused-function warnings.  Declaring them static
inline keeps the same behaviour without the warnings.
---
 include/zbuffer.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/zbuffer.h b/include/zbuffer.h
index 886eb72..c54c9bd 100644
--- a/include/zbuffer.h
+++ b/include/zbuffer.h
@@ -354,9 +354,9 @@ void *gl_zalloc(GLint size);
 #else
 #include<string.h>
 #include<stdlib.h>
-static void gl_free(void* p) { free(p); }
-static void* gl_malloc(GLint size) { return malloc(size); }
-static void* gl_zalloc(GLint size) { return calloc(1, size); }
+static inline void gl_free(void* p) { free(p); }
+static inline void* gl_malloc(GLint size) { return malloc(size); }
+static inline void* gl_zalloc(GLint size) { return calloc(1, size); }
 #endif
 
 #endif /* _tgl_zbuffer_h_ */
-- 
2.43.0

